File: tests/C/0016.c
float foo (int N, float vec[])
{
	int VECLEN = 4;

	#pragma acc parallel loop num_workers(10) copy(vec[:N])
	for (int i = 0; i < N; ++i)
		vec[i] = i + 1.;

	#pragma acc parallel num_gangs(2) copy(vec[:N])
	#pragma acc loop gang
	for (int i = 0; i < N; ++i)
		vec[i] = i + 2.;

	#pragma acc parallel num_gangs(2) num_workers(10) copy(vec[:N])
	#pragma acc loop
	for (int i = 0; i < N; ++i)
		vec[i] = i + 3.;

	#pragma acc parallel num_gangs(2) copy(vec[:N]) if(1==1)
	#pragma acc loop vector
	for (int i = 0; i < N; ++i)
		vec[i] = i + 4.;

	#pragma acc parallel loop vector vector_length(VECLEN)
	for (int i = 0; i < N; ++i)
		vec[i] = i + 4.;

	return vec[N-1];
}

File: tests/Fortran/Fixed/0016.f
      program test
      implicit none
      integer i,j
      real vec(1000)

!$acc parallel loop num_workers(10) copy (vec)
      do i = 1, 1000
        vec(i) = i+1
      end do

!$acc parallel num_gangs(2) copy (vec)
!$acc loop gang
      do i = 1, 1000
        vec(i) = i+2
      end do
!$acc end parallel

!$acc parallel num_workers(10) num_gangs(2) copy (vec)
!$acc loop
      do i = 1, 1000
        vec(i) = i+3
      end do
!$acc end parallel

!$acc parallel num_workers(10) num_gangs(2) copy(vec) if (1.eq.1)
!$acc loop vector
      do i = 1, 1000
        vec(i) = i+4
      end do
!$acc end parallel

!$acc parallel loop vector vector_length(8)
      do i = 1, 1000
        vec(i) = i+5
      end do

!$acc parallel loop
      do i = 1, 1000
!$acc loop seq
        do j = 1, 1000
          vec(i) = i+5
        end do
      end do

!$acc parallel loop
      do i = 1, 1000
!$acc loop
        do j = 1, 1000
          vec(i) = i+5
        end do
      end do

      end program
